python/perf-kernels/06-fused-attention-fwd-transV.py

```diff
@@ -205,34 +207,36 @@ def forward(ctx, q, k, v, sm_scale):
     for shape in [(4, 48, 1024, 128),
                   (4, 48, 2048, 128),
                   (4, 48, 4096, 128)]
-    for dtype in ['fp16', 'fp8']])
+    for dtype in ['fp16', 'bf16', 'fp8']])
 def test_op_fwd(Z, H, N_CTX, D_HEAD, dtype):
     torch.manual_seed(20)
+    init_dtype = torch.float16 if dtype == 'fp8' else name_to_torch_types[dtype]
     q = (
-        torch.empty((Z, H, N_CTX, D_HEAD), dtype=torch.float16, device="cuda")
+        torch.empty((Z, H, N_CTX, D_HEAD), dtype=init_dtype, device="cuda")
         .normal_(mean=0., std=0.5)
         .requires_grad_()
     )
     k = (
-        torch.empty((Z, H, N_CTX, D_HEAD), dtype=torch.float16, device="cuda")
+        torch.empty((Z, H, N_CTX, D_HEAD), dtype=init_dtype, device="cuda")
         .normal_(mean=0., std=0.5)
         .requires_grad_()
     )
     v = (
-        torch.empty((Z, H, D_HEAD, N_CTX), dtype=torch.float16, device="cuda")
+        torch.empty((Z, H, D_HEAD, N_CTX), dtype=init_dtype, device="cuda")
         .normal_(mean=0., std=0.5)
         .requires_grad_()
     )
     sm_scale = 0.5
-    q = q.to(name_to_torch_types[dtype])
-    k = k.to(name_to_torch_types[dtype])
-    dout = torch.randn_like(q, dtype=torch.float16)
     # reference implementation
     M = torch.tril(torch.ones((N_CTX, N_CTX), device="cuda"))
-    p = torch.matmul(q.half(), k.transpose(2, 3).half()) * sm_scale
-    p = torch.softmax(p.float(), dim=-1).half()
+    p = torch.matmul(q, k.transpose(2, 3)) * sm_scale
+    p = torch.softmax(p.float(), dim=-1).to(q.dtype)
     ref_out = torch.matmul(p, v.transpose(2,3))
     # triton implementation
+    # q,k casting for partial fp8
+    q = q.to(name_to_torch_types[dtype])
+    k = k.to(name_to_torch_types[dtype])
+    dout = torch.randn_like(q, dtype=torch.float16)
     tri_out = attention(q, k, v, sm_scale)
     # compare
     atol = 1.4e-1 if dtype == 'fp8' else 1e-2
```

Comment:
Can you loose the threshold for bf16 a little bit like fp8 so that we can have all test pass and be aware that bf16 is not that accurate for now?

Reply:
I don't know what is a good number to set for `atol` and `rtol` s.t. I'm not change the number to "fake" the test pass. I change rtol for non-fp8 to 1e-2 and start using it and now it passes
```
rtol = 1e-2 if dtype == 'fp8' else 1e-2
torch.testing.assert_close(ref_out, tri_out, atol=atol, rtol=rtol)
```

Reply:
This is even better

Reply:
Okay I pushed that in and it's on another commit now.